Assets/Shopify/UIToolkit/Editor/ShopCredentialsVerifier.cs

```diff
@@ -0,0 +1,141 @@
+﻿namespace Shopify.UIToolkit.Editor {
+    using UnityEngine;
+    using UnityEditor;
+    using System;
+    using Shopify.Unity;
+    using Shopify.Unity.SDK;
+    using Shopify.Unity.SDK.Editor;
+
+    /// <summary>
+    /// A helper for the editor that allows verifying shop credentials
+    /// and drawing a UI to support the feature.
+    /// </summary>
+    public class ShopCredentialsVerifier {
+        private IShopCredentials _context;
+        private bool _isRequestInProgress;
+
+        /// <summary>
+        /// Creates a new verifier
+        /// </summary>
+        /// <param name="context">The object with credentials</param>
+        public ShopCredentialsVerifier(IShopCredentials context) {
+            _context = context;
+        }
+
+		/// <summary>
+		/// Draws the GUI for the verifier, should be called from an editor
+		/// in OnInspectorGUI
+		/// </summary>
+		/// <param name="serializedObject">The serialized version of the object provided as the context</param>
+		public void DrawInspectorGUI(SerializedObject serializedObject) {
+            EditorGUILayout.Separator();
+
+            var disableCredentialsForm = _context.CredentialsVerificationState == ShopCredentialsVerificationState.Verified;
+
+            EditorGUI.BeginDisabledGroup(disableCredentialsForm);
+            DrawCredentialsForm(serializedObject);
+            EditorGUI.EndDisabledGroup();
+
+            DrawMessageBox();
+            DrawActionButton();
+            EditorGUILayout.Separator();
+        }
+
+		/// <summary>
+		/// Perform an async request to see if the credentials are valid.
+		/// This request will also mutate the context object's CredentialsVerificationState.
+		/// </summary>
+        /// <param name="onSuccess">Callback called when the credentials are found to be valid</param>
+        /// <param name="onFailure">Callback called when the credentials are found to be invalid</param>
+		public void VerifyCredentials(Action onSuccess, Action onFailure) {
+			if (_isRequestInProgress) 
+				throw new InvalidOperationException("Can't verify credentials when a verification request is already in progress.");
+
+            _isRequestInProgress = true;
+
+            Client().Query((root) => {
+                root.shop((shop) => {
+                    shop.name();
+                });
+            }, (QueryRoot result, ShopifyError error) => {
+                _isRequestInProgress = false;
+
+                OnVerificationRequestComplete(result, error);
+
+                if (error != null) {
+                    onFailure();
+                } else {
+                    onSuccess();
+                }
+            });
+        }
+
+        /// <summary>
+		/// Perform an async request to see if the credentials are valid.
+		/// This request will also mutate the context object's CredentialsVerificationState.
+        /// </summary>
+        public void VerifyCredentials() {
+            VerifyCredentials(() => { }, () => { });
```

Comment:
What's going on here?

Reply:
Just an overload with empty onSuccess/onError callbacks, those are useful for tests/the editor but internally its easier to just call this overload